pkg/connector/client/confluence.go

```diff
@@ -385,6 +385,324 @@ func strToInt(s string) int {
 	return i
 }
 
+// GetSpaces uses pagination to get a list of spaces from the global list.
+func (c *ConfluenceClient) GetSpaces(
+	ctx context.Context,
+	pageSize int,
+	paginationCursor string,
+) (
+	[]ConfluenceSpace,
+	string,
+	*v2.RateLimitDescription,
+	error,
+) {
+	spacesListUrl, err := c.genURLWithPaginationCursor(
+		SpacesListUrlPath,
+		pageSize,
+		paginationCursor,
+	)
+	if err != nil {
+		return nil, "", nil, err
+	}
+
+	var response *confluenceSpaceList
+	ratelimitData, err := c.get(ctx, spacesListUrl, &response)
+	if err != nil {
+		return nil, "", ratelimitData, err
+	}
+
+	cursor := extractPaginationCursor(response.Links)
+	spaces := response.Results
+
+	return spaces, cursor, ratelimitData, nil
+}
+
+func (c *ConfluenceClient) ConfluenceSpaceOperations(
+	ctx context.Context,
+	cursor string,
+	pageSize int,
+	spaceId string,
+) (
+	[]ConfluenceSpaceOperation,
+	string,
+	*v2.RateLimitDescription,
+	error,
+) {
+	logger := ctxzap.Extract(ctx)
+	logger.Debug("fetching space", zap.String("spaceId", spaceId))
+
+	spaceUrl, err := c.genURLWithPaginationCursor(
+		fmt.Sprintf(spacesGetUrlPath+"?include-operations=1", spaceId),
```

Comment:
It'd be nice to avoid building urls with sprintf, but that would require changing genURLWithPaginationCursor() and all its call sites.